src/header/Header.js

```diff
@@ -84,57 +85,66 @@ class Header extends Component {
     } = this.props;
 
     return (
-      <ViewComponent
-        testID="headerContainer"
-        {...attributes}
-        style={StyleSheet.flatten([
-          styles.container(theme),
-          backgroundColor && { backgroundColor },
-          containerStyle,
-        ])}
-        source={backgroundImage}
-        imageStyle={backgroundImageStyle}
-        {...linearGradientProps}
-      >
+      <>
         <StatusBar barStyle={barStyle} translucent={true} {...statusBarProps} />
-        <Children
+        <SafeAreaView
```

Comment:
I think wrapping a reusable component with `SafeAreaView` is not recommended. It must be consumed from the user end. If someone wants to deliberately overflow the Header then, that behaviour can't be achieved.

Reply:
This specific one at least doesn't wrap anything. It has a flex of 0 and is used to work around some issues.
However, without the `SafeAreaView` below wrapping the `Children`, it is almost impossible to get the orientation working as one would expect. The left and right components end up showing up in the curves of the iPhone instead of in the safe area where you'd expect them. (Try removing it and trying it out in the demo app, and see where the left and right components end up)
I'm inclined to think that this covers 99% of use cases and the 1% that want to "overflow" the `Header` will likely be making their own component anyways.

Reply:
I agree. My only concern is that we try to minimize the render tree. People generally use libraries like `react-navigation` which comes with default wrapping of SFA.
https://reactnavigation.org/docs/handling-safe-area/

1. IMO, specifying this workaround in the docs.
2. Or we can add a boolean prop `safeArea` which internally wraps iOS in SafeAreaView and Android in paddingTop status bar height

Reply:
Or a new component `SafeArea ` which does this job. So that it can be used for non-header usage as well. Like apps with bottom nav etc.

Reply:
Alright! I replaced the `SafeAreaView` with a different library. This means that now everything shows up as expected. And Android and iOS look and behave the exact same!